src/httpserver/handlers/handler.clj

```diff
@@ -1,24 +1,27 @@
 (ns httpserver.handlers.handler
-  (:require [httpserver.router :as router]))
+  (:require [httpserver.routing.router :as router]))
 
 (import '(com.td.HttpServer HttpResponse))
 (import '(com.td.HttpServer IHandler))
 (import '(com.td.HttpServer HttpRequest))
 
 (defn build-request [httpRequest]
-  (hash-map :path (.path httpRequest) :method (.method httpRequest) :headers (.headers httpRequest) :body (.body httpRequest)))
+  {:path (.path httpRequest) :method (.method httpRequest) :headers (.headers httpRequest) :body (.body httpRequest)})
+
+(def routes {})
 
 (deftype main-handler []
   com.td.HttpServer.IHandler
   (generateResponse [this httpRequest]
     (let [request (build-request httpRequest)
-          function-to-use (router/get-function request)
+          function-to-use (router/get-handler request routes)
           response (function-to-use request)
           headers (java.util.HashMap. (:headers response))]
       (doto (HttpResponse.)
         (.setBody (:body response))
         (.setResponseCode (:code response))
         (.addHeaders headers)))))
 
-(defn new-handler []
+(defn new-handler [list-of-routes]
+  (def routes list-of-routes)
```

Comment:
Followup on this: Geoff Shannon says that generally speaking you should use `defrecord` over `deftype`. I'm actually pretty rusty on the difference myself—Geoff says that `deftype` is for if you want to implement your own data structures.

Maybe worth researching and blogging?

Also, idiomatically the name should be Java style since they're actually creating Java classes/objects, i.e. `(defrecord MainHandler)` and `(MainHandler. list-of-routes)`